shared/common-adapters/plain-input.js.flow

```diff
@@ -85,11 +86,24 @@ export type TextInfo = {
   selection: Selection,
 }
 
-export type InternalProps = DefaultProps & Props
+export type InternalProps = {...DefaultProps, ...Props}
 declare export default class PlainInput extends React.Component<Props> {
   static defaultProps: DefaultProps;
   blur: () => void;
   focus: () => void;
-  // Supported only on desktop right now
+  /**
+   *  This can only be used when the input is controlled. Use `transformText` if
+   *  you want to do this on an uncontrolled input. Make sure the Selection is
+   *  valid against the `value` prop. Avoid changing `value` and calling this at
```

Comment:
Hmm...so now I'm confused by this API. We almost always want to change both the text and the selection at the same time...AIUI the existing use cases are inserting text (like emoji, mentions, etc.) at the cursor. So that's why I suggested having a `selection` prop along with the value. Did that not work?

Reply:
I ran into issues early on with the parent actively managing the selection and value at the same time. AFAICT on desktop there's no way to listen to events that change the cursor position (without selecting anything). Without the parent knowing the selection beforehand, I think the semantics of the prop would be weird, and even then I'm not sure if there's a way to insert text at the cursor the way we want. I'm beginning to think having a `transformText` that works with controlled inputs would be a good idea, though I'm not sure how to handle that along with updating `props.value`...

Reply:
Ah, yes, I remember running into that when I tried. @chrisnojima did mention there was some way to listen to all DOM events and then filter out the selection ones on a given component, but I didn't know enough to pursue that.

...and then, yeah, you can try implementing `transformText` on a controlled component, but then things get murky w.r.t. the source of truth...

Reply:
@akalin-keybase OK, rather than going down the murky source of truth route, I added a `getSelection` method. I added an example of a controlled input setting both at the same time (`getSelection` -> change `props.value` -> `setSelection`). Let me know what you think